client/quantum_serverless/core/job.py

```diff
@@ -166,11 +168,13 @@ def run(self, program: Program, arguments: Optional[Dict[str, Any]] = None):
         )
         return Job(job_id=job_id, job_client=self)
 
-    def upload(self, program: Program):
+    def upload(self, program: QiskitPattern):
```

Comment:
This can a breaking change so it's a bit tricky but should we call it `pattern`?

Reply:
can you overload functions in python? (i.e. have one that takes a program and mark it deprecated and another that takes a pattern)

Reply:
good question!

Reply:
seems like answer is no (though there are a few workarounds if you search around on the topic...)